pkg/reconciler/taskrun/resources/taskref_test.go

```diff
@@ -667,6 +673,286 @@ func TestGetPipelineFunc_RemoteResolutionInvalidData(t *testing.T) {
 	}
 }
 
+func TestLocalTaskRef_TrustedResourceVerification(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	signer, secretpath, err := ttesting.GetSignerFromFile(ctx, t)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	unsignedTask := ttesting.GetUnsignedTask("test-task")
+	signedTask, err := getSignedTask(unsignedTask, signer, "test-signed")
+	if err != nil {
+		t.Fatal("fail to sign task", err)
+	}
+	// attack another signed task
+	signedTask2, err := getSignedTask(ttesting.GetUnsignedTask("test-task2"), signer, "test-signed2")
```

Comment:
Wondering if we shall separate the "attack" of the signedTask to another function to better demonstrate it is testing "wrong annotation signature will fail the verifcation"?